#  **EDA Dataset 2 and NLP Natural Language Processing** 

Natural Language Processing (NLP) is a field of artificial intelligence (AI) that focuses on the interaction between computers and human language. It involves the development of algorithms and models to enable computers to understand, interpret, and generate human language.

The purpose of this notebook is to convert the reviews of each user into a dicrete column called "sentiment_analysis". The idea is to assign the value of '0' if th review is bad, '1' if it is neutral and '2' if it is positive. If there is not a review, by default we are going to assing '1'.

### 1. Data Load:

In [1]:
import pandas as pd
import json
import ast

# Let's open the file "user_reviews" in .json format and save it into a dataframe called "df_user_reviews"
data = []
with open('Dataset/user_reviews.json', 'r', encoding='utf-8') as file:
    for line in file.readlines():
        data.append(ast.literal_eval(line))

df_user_reviews = pd.DataFrame(data)


### 2. Exploratory Data Analysis (EDA):

In [2]:
df_user_reviews

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1                js41637               http://steamcommunity.com/id/js41637   
2              evcentric             http://steamcommunity.com/id/evcentric   
3                  doctr                 http://steamcommunity.com/id/doctr   
4              maplemage             http://steamcommunity.com/id/maplemage   
...                  ...                                                ...   
25794  76561198306599751  http://steamcommunity.com/profiles/76561198306...   
25795           Ghoustik              http://steamcommunity.com/id/Ghoustik   
25796  76561198310819422  http://steamcommunity.com/profiles/76561198310...   
25797  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                                 reviews  
0      [{'funny': '', 'posted': 'Posted November 5, 2...  
1      [{'funny': '', 'posted': 'Posted June 24, 2014...  
2      [{'funny': '', 'posted': 'Posted February 3.',...  
3      [{'funny': '', 'posted': 'Posted October 14, 2...  
4      [{'funny': '3 people found this review funny',...  
...                                                  ...  
25794  [{'funny': '', 'posted': 'Posted May 31.', 'la...  
25795  [{'funny': '', 'posted': 'Posted June 17.', 'l...  
25796  [{'funny': '1 person found this review funny',...  
25797  [{'funny': '', 'posted': 'Posted July 21.', 'l...  
25798  [{'funny': '1 person found this review funny',...  

[25799 rows x 3 columns]

In [3]:
df_user_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25799 entries, 0 to 25798
Data columns (total 3 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   user_id   25799 non-null  object
 1   user_url  25799 non-null  object
 2   reviews   25799 non-null  object
dtypes: object(3)
memory usage: 604.8+ KB


We can see that this dataframe has 25,799 entries and 3 columns. Also, we don't have nulls that need to be addressed.

Our first sight is that we have different reviews per user_id. Also, theres is a user_url per user_id. By this way, we can observe that the column "reviews" has a list of dictionaries. Let's check the keys of the dictionaries:

In [4]:
df_user_reviews.iloc[0,2]

[{'funny': '',
  'posted': 'Posted November 5, 2011.',
  'last_edited': '',
  'item_id': '1250',
  'helpful': 'No ratings yet',
  'recommend': True,
  'review': 'Simple yet with great replayability. In my opinion does "zombie" hordes and team work better than left 4 dead plus has a global leveling system. Alot of down to earth "zombie" splattering fun for the whole family. Amazed this sort of FPS is so rare.'},
 {'funny': '',
  'posted': 'Posted July 15, 2011.',
  'last_edited': '',
  'item_id': '22200',
  'helpful': 'No ratings yet',
  'recommend': True,
  'review': "It's unique and worth a playthrough."},
 {'funny': '',
  'posted': 'Posted April 21, 2011.',
  'last_edited': '',
  'item_id': '43110',
  'helpful': 'No ratings yet',
  'recommend': True,
  'review': 'Great atmosphere. The gunplay can be a bit chunky at times but at the end of the day this game is definitely worth it and I hope they do a sequel...so buy the game so I get a sequel!'}]

We can see that we have 7 keys in the dictionaries. The idea is to convert those keys into columns of the dataframe. 

### 3. Data Transformation:

We need to separate the list in the column "reviews" for each user_id. What we are going to do is to expand each user_id row with each element of the list. For this purpose, we use the function "explode".

In [5]:
df_user_reviews_exploded = df_user_reviews.explode('reviews')
df_user_reviews_exploded

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1                js41637               http://steamcommunity.com/id/js41637   
1                js41637               http://steamcommunity.com/id/js41637   
...                  ...                                                ...   
25797  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
25797  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
25798        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                                 reviews  
0      {'funny': '', 'posted': 'Posted November 5, 20...  
0      {'funny': '', 'posted': 'Posted July 15, 2011....  
0      {'funny': '', 'posted': 'Posted April 21, 2011...  
1      {'funny': '', 'posted': 'Posted June 24, 2014....  
1      {'funny': '', 'posted': 'Posted September 8, 2...  
...                                                  ...  
25797  {'funny': '', 'posted': 'Posted July 10.', 'la...  
25797  {'funny': '', 'posted': 'Posted July 8.', 'las...  
25798  {'funny': '1 person found this review funny', ...  
25798  {'funny': '', 'posted': 'Posted July 20.', 'la...  
25798  {'funny': '', 'posted': 'Posted July 2.', 'las...  

[59333 rows x 3 columns]

we can check that the user_id rows were expanded according to the number of list elements. For instance, the user_id 76561197970982479 was expanded three times.

Now we need to convert the keys of each ditionary into other columns in the dataframe. For this purpose, we are going to use the function json_normalize for Pandas as follows:

In [6]:
df_user_reviews_normalized = pd.json_normalize(df_user_reviews_exploded['reviews'])
df_user_reviews_normalized

funny                     posted  \
0                                         Posted November 5, 2011.   
1                                            Posted July 15, 2011.   
2                                           Posted April 21, 2011.   
3                                            Posted June 24, 2014.   
4                                        Posted September 8, 2013.   
...                                 ...                        ...   
59328                                              Posted July 10.   
59329                                               Posted July 8.   
59330  1 person found this review funny             Posted July 3.   
59331                                              Posted July 20.   
59332                                               Posted July 2.   

      last_edited item_id                                          helpful  \
0                    1250                                   No ratings yet   
1                   22200                                   No ratings yet   
2                   43110                                   No ratings yet   
3                  251610  15 of 20 people (75%) found this review helpful   
4                  227300     0 of 1 people (0%) found this review helpful   
...           ...     ...                                              ...   
59328                  70                                   No ratings yet   
59329              362890                                   No ratings yet   
59330              273110    1 of 2 people (50%) found this review helpful   
59331                 730                                   No ratings yet   
59332                 440                                   No ratings yet   

      recommend                                             review  
0          True  Simple yet with great replayability. In my opi...  
1          True               It's unique and worth a playthrough.  
2          True  Great atmosphere. The gunplay can be a bit chu...  
3          True  I know what you think when you see this title ...  
4          True  For a simple (it's actually not all that simpl...  
...         ...                                                ...  
59328      True  a must have classic from steam definitely wort...  
59329      True  this game is a perfect remake of the original ...  
59330      True  had so much fun plaing this and collecting res...  
59331      True                                                 :D  
59332      True                                     so much fun :D  

[59333 rows x 7 columns]

Now, with the "reviews" column transformed into a Dataframe, we need to concatenate it with "df_user_reviews_exploded" using the fuunction concat from Pandas.

In [7]:
# We need to get rid of the indexes in both dataframes first:

df_user_reviews_exploded.reset_index(drop=True, inplace=True)
df_user_reviews_normalized.reset_index(drop=True, inplace=True)

In [8]:
#Concat both dataframes:
df_user_reviews_expanded = pd.concat([df_user_reviews_exploded.drop('reviews',axis=1),df_user_reviews_normalized], axis=1)
df_user_reviews_expanded

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
2      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
3                js41637               http://steamcommunity.com/id/js41637   
4                js41637               http://steamcommunity.com/id/js41637   
...                  ...                                                ...   
59328  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59329  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59330        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59331        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59332        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                  funny                     posted  \
0                                         Posted November 5, 2011.   
1                                            Posted July 15, 2011.   
2                                           Posted April 21, 2011.   
3                                            Posted June 24, 2014.   
4                                        Posted September 8, 2013.   
...                                 ...                        ...   
59328                                              Posted July 10.   
59329                                               Posted July 8.   
59330  1 person found this review funny             Posted July 3.   
59331                                              Posted July 20.   
59332                                               Posted July 2.   

      last_edited item_id                                          helpful  \
0                    1250                                   No ratings yet   
1                   22200                                   No ratings yet   
2                   43110                                   No ratings yet   
3                  251610  15 of 20 people (75%) found this review helpful   
4                  227300     0 of 1 people (0%) found this review helpful   
...           ...     ...                                              ...   
59328                  70                                   No ratings yet   
59329              362890                                   No ratings yet   
59330              273110    1 of 2 people (50%) found this review helpful   
59331                 730                                   No ratings yet   
59332                 440                                   No ratings yet   

      recommend                                             review  
0          True  Simple yet with great replayability. In my opi...  
1          True               It's unique and worth a playthrough.  
2          True  Great atmosphere. The gunplay can be a bit chu...  
3          True  I know what you think when you see this title ...  
4          True  For a simple (it's actually not all that simpl...  
...         ...                                                ...  
59328      True  a must have classic from steam definitely wort...  
59329      True  this game is a perfect remake of the original ...  
59330      True  had so much fun plaing this and collecting res...  
59331      True                                                 :D  
59332      True                                     so much fun :D  

[59333 rows x 9 columns]

In [24]:
df_user_reviews_expanded.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 59333 entries, 0 to 59332
Data columns (total 9 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   user_id      59333 non-null  object
 1   user_url     59333 non-null  object
 2   funny        59305 non-null  object
 3   posted       59305 non-null  object
 4   last_edited  59305 non-null  object
 5   item_id      59305 non-null  object
 6   helpful      59305 non-null  object
 7   recommend    59305 non-null  object
 8   review       59305 non-null  object
dtypes: object(9)
memory usage: 4.1+ MB


Finally, we have unnested the initial dataframe into something that is easier to manipulate. Now, let's move to  application of NLP.

### 4. Sentiment Analysis:

To do a sentiment analysis of a text we can use the library [vaderSentiment](https://github.com/cjhutto/vaderSentiment). 

The VADER (Valence Aware Dictionary and sEntiment Reasoner) library is a tool for analyzing and quantifying the sentiment or emotional tone of text data. It is specifically designed to work with text data that is often found in social media, online reviews, and informal communication, where traditional sentiment analysis methods may struggle due to the use of slang, abbreviations, emojis, and other language nuances.

It has a function that is called "sentiment.polarity_scores". It can be programmed as follows:

In [51]:
#Import the library
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer

In [92]:
sentiment = SentimentIntensityAnalyzer()
text_1 = "The book was a perfect balance between wrtiting style and plot."
text_2 =  "The pizza tastes terrible."
sent_1 = sentiment.polarity_scores(text_1)
sent_2 = sentiment.polarity_scores(text_2)
print("Sentiment of text 1:", sent_1)
print("Sentiment of text 2:", sent_2)

Sentiment of text 1: {'neg': 0.0, 'neu': 0.73, 'pos': 0.27, 'compound': 0.5719}
Sentiment of text 2: {'neg': 0.508, 'neu': 0.492, 'pos': 0.0, 'compound': -0.4767}


According to the [documentation](https://github.com/cjhutto/vaderSentiment), this is the interpretation of the scoring:

* The ``compound`` score is computed by summing the valence scores of each word in the lexicon, adjusted according to the rules, and then normalized to be between -1 (most extreme negative) and +1 (most extreme positive). This is the most useful metric if you want a single unidimensional measure of sentiment for a given sentence. Calling it a 'normalized, weighted composite score' is accurate. 
 
  It is also useful for researchers who would like to set standardized thresholds for classifying sentences as either positive, neutral, or negative.  
  Typical threshold values (used in the literature cited on this page) are:

  - **positive sentiment**: ``compound`` score >=  0.05
  - **neutral  sentiment**: (``compound`` score > -0.05) and (``compound`` score < 0.05)
  - **negative sentiment**: ``compound`` score <= -0.05
 
  **NOTE:** The ``compound`` score is the one most commonly used for sentiment analysis by most researchers, including the authors.

* The ``pos``, ``neu``, and ``neg`` scores are *ratios for proportions of text that fall in each category* (so these should all add up to be 1... or close to it with float operation).  These are the most useful metrics if you want to analyze the context & presentation of how sentiment is conveyed or embedded in rhetoric for a given sentence. For example, different writing styles may embed strongly positive or negative sentiment within varying proportions of neutral text -- i.e., some writing styles may reflect a penchant for strongly flavored rhetoric, whereas other styles may use a great deal of neutral text while still conveying a similar overall (compound) sentiment. As another example: researchers analyzing information presentation in journalistic or editorical news might desire to establish whether the proportions of text (associated with a topic or named entity, for example) are balanced with similar amounts of positively and negatively framed text versus being "biased" towards one polarity or the other for the topic/entity.

  * **IMPORTANTLY:** these proportions represent the "raw categorization" of each lexical item (e.g., words, emoticons/emojis, or initialisms) into positve, negative, or neutral classes; they **do not** account for the VADER rule-based enhancements such as word-order sensitivity for sentiment-laden multi-word phrases, degree modifiers, word-shape amplifiers, punctuation amplifiers, negation polarity switches, or contrastive conjunction sensitivity.


Now that we have understood how we can leverage this library for our analysis, let's try to use it with our first review in the dataframe:

In [101]:
# Let's do the exercise with the first row of reviews
testSA = df_user_reviews_expanded.iloc[0,8]
text = testSA
text

'Simple yet with great replayability. In my opinion does "zombie" hordes and team work better than left 4 dead plus has a global leveling system. Alot of down to earth "zombie" splattering fun for the whole family. Amazed this sort of FPS is so rare.'

In [102]:
# We obtain the sentiment scores
sent_1 = sentiment.polarity_scores(text)
print("Sentiment of text:", sent_1)

Sentiment of text: {'neg': 0.074, 'neu': 0.692, 'pos': 0.234, 'compound': 0.8481}


In [110]:
# We are going to use the compound score to classify the review in the categories 0,1 and 2. We can found this value
# in the 'compound' key of the dictionary
sent_1['compound']

0.8481

Now it's time to do the same for all the dataframe:

In [114]:
sentiment_score = []
for text in df_user_reviews_expanded['review']:
    if isinstance(text, str):
        sent = sentiment.polarity_scores(text)
        sentiment_score.append(sent['compound'])
    else:
        # Handle cases where 'review' is not a string (e.g., NaN values)
        sentiment_score.append(0.0)  


sentiment_score[:5]

[0.8481, 0.2263, 0.9117, 0.9899, 0.9958]

Let's create a copy of our original expanded data set in order to add the column called "sentiment_score":

In [115]:
df_user_reviews_expanded1 = df_user_reviews_expanded.copy()
df_user_reviews_expanded1 

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
2      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
3                js41637               http://steamcommunity.com/id/js41637   
4                js41637               http://steamcommunity.com/id/js41637   
...                  ...                                                ...   
59328  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59329  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59330        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59331        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59332        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                  funny                     posted  \
0                                         Posted November 5, 2011.   
1                                            Posted July 15, 2011.   
2                                           Posted April 21, 2011.   
3                                            Posted June 24, 2014.   
4                                        Posted September 8, 2013.   
...                                 ...                        ...   
59328                                              Posted July 10.   
59329                                               Posted July 8.   
59330  1 person found this review funny             Posted July 3.   
59331                                              Posted July 20.   
59332                                               Posted July 2.   

      last_edited item_id                                          helpful  \
0                    1250                                   No ratings yet   
1                   22200                                   No ratings yet   
2                   43110                                   No ratings yet   
3                  251610  15 of 20 people (75%) found this review helpful   
4                  227300     0 of 1 people (0%) found this review helpful   
...           ...     ...                                              ...   
59328                  70                                   No ratings yet   
59329              362890                                   No ratings yet   
59330              273110    1 of 2 people (50%) found this review helpful   
59331                 730                                   No ratings yet   
59332                 440                                   No ratings yet   

      recommend                                             review  
0          True  Simple yet with great replayability. In my opi...  
1          True               It's unique and worth a playthrough.  
2          True  Great atmosphere. The gunplay can be a bit chu...  
3          True  I know what you think when you see this title ...  
4          True  For a simple (it's actually not all that simpl...  
...         ...                                                ...  
59328      True  a must have classic from steam definitely wort...  
59329      True  this game is a perfect remake of the original ...  
59330      True  had so much fun plaing this and collecting res...  
59331      True                                                 :D  
59332      True                                     so much fun :D  

[59333 rows x 9 columns]

In [116]:
df_user_reviews_expanded1['sentiment_score'] = sentiment_score

In [117]:
df_user_reviews_expanded1

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
2      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
3                js41637               http://steamcommunity.com/id/js41637   
4                js41637               http://steamcommunity.com/id/js41637   
...                  ...                                                ...   
59328  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59329  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59330        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59331        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59332        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                  funny                     posted  \
0                                         Posted November 5, 2011.   
1                                            Posted July 15, 2011.   
2                                           Posted April 21, 2011.   
3                                            Posted June 24, 2014.   
4                                        Posted September 8, 2013.   
...                                 ...                        ...   
59328                                              Posted July 10.   
59329                                               Posted July 8.   
59330  1 person found this review funny             Posted July 3.   
59331                                              Posted July 20.   
59332                                               Posted July 2.   

      last_edited item_id                                          helpful  \
0                    1250                                   No ratings yet   
1                   22200                                   No ratings yet   
2                   43110                                   No ratings yet   
3                  251610  15 of 20 people (75%) found this review helpful   
4                  227300     0 of 1 people (0%) found this review helpful   
...           ...     ...                                              ...   
59328                  70                                   No ratings yet   
59329              362890                                   No ratings yet   
59330              273110    1 of 2 people (50%) found this review helpful   
59331                 730                                   No ratings yet   
59332                 440                                   No ratings yet   

      recommend                                             review  \
0          True  Simple yet with great replayability. In my opi...   
1          True               It's unique and worth a playthrough.   
2          True  Great atmosphere. The gunplay can be a bit chu...   
3          True  I know what you think when you see this title ...   
4          True  For a simple (it's actually not all that simpl...   
...         ...                                                ...   
59328      True  a must have classic from steam definitely wort...   
59329      True  this game is a perfect remake of the original ...   
59330      True  had so much fun plaing this and collecting res...   
59331      True                                                 :D   
59332      True                                     so much fun :D   

       sentiment_score  
0               0.8481  
1               0.2263  
2               0.9117  
3               0.9899  
4               0.9958  
...                ...  
59328           0.5574  
59329           0.9786  
59330           0.7827  
59331           0.5106  
59332           0.8349  

[59333 rows x 10 columns]

We can use the "sentiment_score" to do the final classification into 0,1,2 considering the following rules as the documentation suggests:

  - **positive sentiment**: ``compound`` score >=  0.05
  - **neutral  sentiment**: (``compound`` score > -0.05) and (``compound`` score < 0.05)
  - **negative sentiment**: ``compound`` score <= -0.05

In [118]:
sentiment_analysis = []
for score in df_user_reviews_expanded1['sentiment_score']:
    if score >= 0.05:
        sent = 2
    elif score > -0.05 and score < 0.05:
        sent = 1
    else:
        sent = 0
    sentiment_analysis.append(sent)

In [120]:
df_user_reviews_expanded1['sentiment_analysis'] = sentiment_analysis
df_user_reviews_expanded1

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
2      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
3                js41637               http://steamcommunity.com/id/js41637   
4                js41637               http://steamcommunity.com/id/js41637   
...                  ...                                                ...   
59328  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59329  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59330        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59331        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59332        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                  funny                     posted  \
0                                         Posted November 5, 2011.   
1                                            Posted July 15, 2011.   
2                                           Posted April 21, 2011.   
3                                            Posted June 24, 2014.   
4                                        Posted September 8, 2013.   
...                                 ...                        ...   
59328                                              Posted July 10.   
59329                                               Posted July 8.   
59330  1 person found this review funny             Posted July 3.   
59331                                              Posted July 20.   
59332                                               Posted July 2.   

      last_edited item_id                                          helpful  \
0                    1250                                   No ratings yet   
1                   22200                                   No ratings yet   
2                   43110                                   No ratings yet   
3                  251610  15 of 20 people (75%) found this review helpful   
4                  227300     0 of 1 people (0%) found this review helpful   
...           ...     ...                                              ...   
59328                  70                                   No ratings yet   
59329              362890                                   No ratings yet   
59330              273110    1 of 2 people (50%) found this review helpful   
59331                 730                                   No ratings yet   
59332                 440                                   No ratings yet   

      recommend                                             review  \
0          True  Simple yet with great replayability. In my opi...   
1          True               It's unique and worth a playthrough.   
2          True  Great atmosphere. The gunplay can be a bit chu...   
3          True  I know what you think when you see this title ...   
4          True  For a simple (it's actually not all that simpl...   
...         ...                                                ...   
59328      True  a must have classic from steam definitely wort...   
59329      True  this game is a perfect remake of the original ...   
59330      True  had so much fun plaing this and collecting res...   
59331      True                                                 :D   
59332      True                                     so much fun :D   

       sentiment_score  sentiment_analysis  
0               0.8481                   2  
1               0.2263                   2  
2               0.9117                   2  
3               0.9899                   2  
4               0.9958                   2  
...                ...                 ...  
59328           0.5574                   2  
59329           0.9786                   2  
59330           0.7827                   2  

In [121]:
columns_to_eliminate = ['review','sentiment_score']
df_user_reviews_final = df_user_reviews_expanded1.drop(columns=columns_to_eliminate)
df_user_reviews_final

user_id                                           user_url  \
0      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
1      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
2      76561197970982479  http://steamcommunity.com/profiles/76561197970...   
3                js41637               http://steamcommunity.com/id/js41637   
4                js41637               http://steamcommunity.com/id/js41637   
...                  ...                                                ...   
59328  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59329  76561198312638244  http://steamcommunity.com/profiles/76561198312...   
59330        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59331        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   
59332        LydiaMorley           http://steamcommunity.com/id/LydiaMorley   

                                  funny                     posted  \
0                                         Posted November 5, 2011.   
1                                            Posted July 15, 2011.   
2                                           Posted April 21, 2011.   
3                                            Posted June 24, 2014.   
4                                        Posted September 8, 2013.   
...                                 ...                        ...   
59328                                              Posted July 10.   
59329                                               Posted July 8.   
59330  1 person found this review funny             Posted July 3.   
59331                                              Posted July 20.   
59332                                               Posted July 2.   

      last_edited item_id                                          helpful  \
0                    1250                                   No ratings yet   
1                   22200                                   No ratings yet   
2                   43110                                   No ratings yet   
3                  251610  15 of 20 people (75%) found this review helpful   
4                  227300     0 of 1 people (0%) found this review helpful   
...           ...     ...                                              ...   
59328                  70                                   No ratings yet   
59329              362890                                   No ratings yet   
59330              273110    1 of 2 people (50%) found this review helpful   
59331                 730                                   No ratings yet   
59332                 440                                   No ratings yet   

      recommend  sentiment_analysis  
0          True                   2  
1          True                   2  
2          True                   2  
3          True                   2  
4          True                   2  
...         ...                 ...  
59328      True                   2  
59329      True                   2  
59330      True                   2  
59331      True                   2  
59332      True                   2  

[59333 rows x 9 columns]

### 5. Export the final DataFrame to a CSV file:

In [122]:
df_user_reviews_final.to_csv('user_reviews_final.csv', index=False)